core/html_api.php

```diff
@@ -1390,10 +1390,22 @@ function html_status_legend( $p_display_position, $p_restrict_by_filter = false
 	$t_workflow = config_get( 'status_enum_workflow' );
 	if( !empty( $t_workflow ) ) {
 		foreach( $t_status_array as $t_status => $t_name ) {
+			# Check if the workflow allows leaving the state
 			if( !isset( $t_workflow[$t_status] ) ) {
 
+				# Check if the workflow allows entering the state
+				$t_can_enter = false;
+				foreach( $t_workflow as $t_values ) {
+					if( MantisEnum::hasValue( $t_values, $t_status ) ) {
+						$t_can_enter = true;
+						break;
+					}
+				}
+
 				# drop elements that are not in the workflow
-				unset( $t_status_array[$t_status] );
+				if( !$t_can_enter ) {
+					unset( $t_status_array[$t_status] );
```

Comment:
Not saying this is right, but I've always seen this done that way in PHP. As far as I know, this piece of code has been working just fine like this since its introduction 2008 (d953981e59743ddf6b6ccbc9c3838353ee25b648). 

Moving it out of the loop would require setting up a temp array storing statuses to unset, and iterating over it after the main loop. This adds complexity to the code, without immediate benefit (if it ain't broke...)

Are you aware of a case where this logic causes issues ?